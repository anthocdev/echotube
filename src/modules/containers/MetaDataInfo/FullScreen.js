import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import CloseIcon from "@material-ui/icons/Close";
import Slide from "@material-ui/core/Slide";
import MetaDataInfo from "./MetaDataInfo";

const useStyles = makeStyles((theme) => ({
  appBar: {
    position: "relative",
  },
  title: {
    marginLeft: theme.spacing(2),
    flex: 1,
  },
}));

const TransitionEffect = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

export default function FullScreen(props) {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const { video } = props;
  return (
    <div>
      <Button
        variant="contained"
        style={{ backgroundColor: "white" }}
        onClick={handleClickOpen}
      >
        Display Meta-Data
      </Button>
      <Dialog
        fullScreen
        open={open}
        onClose={handleClose}
        TransitionComponent={TransitionEffect}
      >
        <AppBar
          style={{ background: "rgba(153, 50, 204, 0.9)" }}
          className={classes.appBar}
        >
          <Toolbar>
            <IconButton
              edge="start"
              color="inherit"
              onClick={handleClose}
              aria-label="close"
            >
              <CloseIcon />
            </IconButton>
            <Typography variant="h6" className={classes.title}>
              {video.Name}
            </Typography>
          </Toolbar>
        </AppBar>

        <MetaDataInfo video={video} onOpen={handleClickOpen} />
      </Dialog>
    </div>
  );
}
