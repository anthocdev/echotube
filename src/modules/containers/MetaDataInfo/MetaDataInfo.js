import React from "react";
import {
  List,
  ListItem,
  ListItemText,
  Typography,
  Divider,
  ListItemAvatar,
  Button,
} from "@material-ui/core";
import DeezerDialog from "../Forms/DeezerManagement/DeezerDialog";
import * as MetaDataActions from "../../../actions/MetaDataActions";
import { connect } from "react-redux";

class MetaDataInfo extends React.Component {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    this.props.onOpen(
      this.props.getMetadata(this.props.video.playlistvideo.PlaylistVideoID)
    );
  }
  render() {
    const deezerSerach = (
      <ListItem button>
        <DeezerDialog video={this.props.video} />
      </ListItem>
    );
    const { Name, Description, ChannelName } = this.props.video;
    const videoDetails = (
      <div>
        <Typography variant="h4" component="h4" style={{ textAlign: "center" }}>
          Video Information
        </Typography>
        <ListItem>
          <ListItemText
            primary="YouTube Channel Name"
            secondary={ChannelName}
          />
        </ListItem>
        <ListItem>
          <ListItemText
            primary="Stored Video Description"
            secondary={Description}
          />
        </ListItem>
      </div>
    );
    if (!this.props.metaData)
      return (
        <List>
          {deezerSerach}
          <Divider />
          {videoDetails}
          <div>No Metadata Stored</div>
        </List>
      );
    {
      /* Plling props only after metadata is checked to exist. */
    }
    const {
      deezerid,
      song_title,
      preview,
      artist_name,
      artist_image,
      album_name,
      album_image,
    } = this.props.metaData;
    return (
      <List>
        {deezerSerach}
        <Divider />
        {videoDetails}
        <div style={{ display: "grid" }}>
          <Typography
            variant="h4"
            component="h4"
            style={{ textAlign: "center" }}
          >
            Deezer Metadata
          </Typography>
          <Button
            variant="contained"
            onClick={() =>
              window.open(
                `https://www.deezer.com/en/track/${deezerid}`,
                "_blank"
              )
            }
          >
            Open Deezer Page
          </Button>
        </div>

        <ListItem>
          <ListItemText primary="Song Title" secondary={song_title} />
        </ListItem>
        <ListItem>
          <Typography variant="h5" component="h5">
            Preview
          </Typography>
          {<audio controls src={preview} style={{ paddingLeft: "20px" }} />}
        </ListItem>
        <ListItem alignItems="flex-start">
          {/* Artist */}
          <ListItem alignItems="flex-start">
            <ListItemAvatar>
              <img src={artist_image} />
            </ListItemAvatar>
            <ListItemText
              primary="Artist"
              secondary={artist_name}
              style={{ paddingLeft: "20px" }}
            />
          </ListItem>
          {/* Album */}
          <ListItem alignItems="flex-start">
            <ListItemAvatar>
              <img src={album_image} />
            </ListItemAvatar>
            <ListItemText
              primary="Album"
              secondary={album_name}
              style={{ paddingLeft: "20px" }}
            />
          </ListItem>
        </ListItem>
      </List>
    );
  }
}

const mapStateToProps = (store) => {
  return {
    metaData: store.metaData.Metadata.metadatum,
  };
};

export default connect(mapStateToProps, MetaDataActions)(MetaDataInfo);
