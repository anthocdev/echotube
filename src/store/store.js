import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
