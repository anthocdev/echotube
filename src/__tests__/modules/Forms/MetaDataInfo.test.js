import React from "react";
import MetaDataInfo from "../../../modules/containers/MetaDataInfo/MetaDataInfo";
import { Provider } from "react-redux";
import store from "../../../store/store";
import { create, act } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

/* Example Video Object for Integration Testing with API Data Structure */
const data = {
  VideoID: 27,
  Name: "Castion - El Zorro",
  Description:
    "[Electro House] Castion - El Zorro (Extended Mix) ➥ Our Spotify: https://illumi.fanlink.to/spotify 📱 Subscribe! Help us to 250,000 🔥 Turn on notifications to stay updated! 🔔 El Zorro by Castion is out now on Mixmash Bold! 🔥 ⬇️ Stream/Download: http://bit.ly/ELZORRO 🎶 Our Spotify Playlists: Future House 🕺 https://illumi.fanlink.to/futurehouse Future Bounce 🔥 https://illumi.fanlink.to/futurebounce Bass House 😈 https://illumi.fanlink.to/basshouse Deep House 🌴 https://illumi.fanlink.to/deephouse YouTube Uploads ▶️ https://illumi.fanlink.to/youtubeuploads 🙏 Join to Illumi Nation Discord: https://discord.gg/Qd8KXru 👕 Merch: http://bit.ly/illumimerch 🎧 Submit Your Track: https://submissions.illumimusic.eu 📀 Promo for Record Labels: promo@illumimusic.eu 📧 Contact: contact@illumimusic.eu ©️🚫This track is NOT copyright free! Copyright Free Music here: http://bit.ly/2si4AyM ➥ Follow Illumi Music: https://www.facebook.com/IllumiMusic https://soundcloud.com/illumi-music https://twitter.com/IllumiMusic https://www.instagram.com/illumimusicyt https://www.snapchat.com/add/illumimusic https://vk.com/illumimusictv ➥ Follow Castion: https://soundcloud.com/castionmusic https://www.facebook.com/CastionMusic https://twitter.com/CastionMusic https://www.instagram.com/castionmusic 📷 Photo by Philipp Lansing Tags: #Castion #ElZorro #BassHouse #Mixmash #FutureHouse #FutureBounce #HouseMusic #DeepHouse #ElectroHouse #EDM #PartyMusic #ShuffleDance #Shuffle #FestivalMusic #Electronic #ElectronicMusic #Dance #Dancemusic #Summermusic #Music #NewMusic #ElectronicDanceMusic #IllumiMusic #Illumi #Music #Future #House Castion - El Zorro Castion - El Zorro (Extended Mix)",
  ChannelName: "Illumi Music",
  ChannelID: "UCXarSAuECvnQFBb1vMcMsjQ",
  VideoLink: "DZjpDfgXNe0",
  playlistvideo: { PlaylistVideoID: 1 },
};

//Mock function to avoid undefined functions
const handleClickOpenMock = () => {};

describe("Metadata Information Module Testing", () => {
  let root;

  it("Renders Without Crashing", async () => {
    act(() => {
      root = create(
        <Provider store={store}>
          <MetaDataInfo video={data} onOpen={handleClickOpenMock} />
        </Provider>
      );
    });
  });

  test("Renders Matches Snapshot", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <MetaDataInfo video={data} onOpen={handleClickOpenMock} />
          </Provider>
        </BrowserRouter>
      );
    });
    expect(root).toMatchSnapshot();
    console.log(root.toJSON());
  });
});
