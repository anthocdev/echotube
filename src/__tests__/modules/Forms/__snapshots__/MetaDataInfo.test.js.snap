// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metadata Information Module Testing Renders Matches Snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Find on Deezer
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <span
      className="MuiTouchRipple-root"
    />
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div>
    <h4
      className="MuiTypography-root MuiTypography-h4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Video Information
    </h4>
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          YouTube Channel Name
        </span>
        <p
          className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          Illumi Music
        </p>
      </div>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Stored Video Description
        </span>
        <p
          className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          [Electro House] Castion - El Zorro (Extended Mix) ➥ Our Spotify: https://illumi.fanlink.to/spotify 📱 Subscribe! Help us to 250,000 🔥 Turn on notifications to stay updated! 🔔 El Zorro by Castion is out now on Mixmash Bold! 🔥 ⬇️ Stream/Download: http://bit.ly/ELZORRO 🎶 Our Spotify Playlists: Future House 🕺 https://illumi.fanlink.to/futurehouse Future Bounce 🔥 https://illumi.fanlink.to/futurebounce Bass House 😈 https://illumi.fanlink.to/basshouse Deep House 🌴 https://illumi.fanlink.to/deephouse YouTube Uploads ▶️ https://illumi.fanlink.to/youtubeuploads 🙏 Join to Illumi Nation Discord: https://discord.gg/Qd8KXru 👕 Merch: http://bit.ly/illumimerch 🎧 Submit Your Track: https://submissions.illumimusic.eu 📀 Promo for Record Labels: promo@illumimusic.eu 📧 Contact: contact@illumimusic.eu ©️🚫This track is NOT copyright free! Copyright Free Music here: http://bit.ly/2si4AyM ➥ Follow Illumi Music: https://www.facebook.com/IllumiMusic https://soundcloud.com/illumi-music https://twitter.com/IllumiMusic https://www.instagram.com/illumimusicyt https://www.snapchat.com/add/illumimusic https://vk.com/illumimusictv ➥ Follow Castion: https://soundcloud.com/castionmusic https://www.facebook.com/CastionMusic https://twitter.com/CastionMusic https://www.instagram.com/castionmusic 📷 Photo by Philipp Lansing Tags: #Castion #ElZorro #BassHouse #Mixmash #FutureHouse #FutureBounce #HouseMusic #DeepHouse #ElectroHouse #EDM #PartyMusic #ShuffleDance #Shuffle #FestivalMusic #Electronic #ElectronicMusic #Dance #Dancemusic #Summermusic #Music #NewMusic #ElectronicDanceMusic #IllumiMusic #Illumi #Music #Future #House Castion - El Zorro Castion - El Zorro (Extended Mix)
        </p>
      </div>
    </li>
  </div>
  <div>
    No Metadata Stored
  </div>
</ul>
`;
